src/Rector/AbstractRector.php

```diff
@@ -83,6 +85,19 @@ abstract class AbstractRector extends NodeVisitorAbstract implements RectorInter
 
     private ?int $toBeRemovedNodeId = null;
 
+    /**
+     * @var array<string, object>
+     */
+    private array $deprecatedDependencies = [];
+
+    /**
+     * Handle deprecated dependencies compatbility
+     */
+    public function __get(string $name): mixed
+    {
```

Comment:
I think this should has soft warning:

```php
        if (! property_exists($this, $name) && isset($this->deprecatedDependencies[$name])) {
            $this->symfonyStyle->warning(
                sprintf('pull %s property from AbstractRector is deprecated, inject via __construct() instead', $name)
            );
       }
```

Reply:
Good idea :+1:  Could you add it?

Reply:
I wil try 👍

Reply:
adding warning on `__get()` seems make too many verbose in console:

```
➜  rector-src git:(main) ✗ bin/rector process src 
   0/146 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%

pull phpDocInfoFactory property from AbstractRector on class Rector\PHPUnit\CodeQuality\Rector\Class_\AddSeeTestAnnotationRector is deprecated, inject via __construct() instead
pull phpDocInfoFactory property from AbstractRector on class Rector\PHPUnit\CodeQuality\Rector\Class_\AddSeeTestAnnotationRector is deprecated, inject via __construct() instead
pull phpDocInfoFactory property from AbstractRector on class Rector\PHPUnit\CodeQuality\Rector\Class_\AddSeeTestAnnotationRector is deprecated, inject via __construct() instead
pull phpDocInfoFactory property from AbstractRector on class Rector\PHPUnit\AnnotationsToAttributes\Rector\Class_\TicketAnnotationToAttributeRector is deprecated, inject via __construct() instead
pull phpDocInfoFactory property from AbstractRector on class Rector\PHPUnit\AnnotationsToAttributes\Rector\Class_\AnnotationWithValueToAttributeRector is deprecated, inject via __construct() instead
```

if next release is 0.19, then I think the `__get()` itself is unnecessary, then add release note instead.

Reply:
Ok, message seems can be cached, see https://github.com/rectorphp/rector-src/pull/5069